src/ORM/Connect/DuplicateEntryException.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace SilverStripe\ORM\Connect;
+
+/**
+ * Exception for errors related to duplicate entries (e.g. violating a unique index)
+ */
+class DuplicateEntryException extends DatabaseException
+{
+    private ?string $keyName = null;
+
+    private ?string $duplicatedValue = null;
```

Comment:
Although the value isn't used in our catch, this info could be useful when debugging.